app/src/test/java/org/simple/clinic/settings/SettingsEffectHandlerTest.kt

```diff
@@ -0,0 +1,80 @@
+package org.simple.clinic.settings
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Observable
+import io.reactivex.observers.TestObserver
+import io.reactivex.subjects.PublishSubject
+import org.junit.After
+import org.junit.Test
+import org.simple.clinic.patient.PatientMocker
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.None
+import org.simple.clinic.util.scheduler.TrampolineSchedulersProvider
+import org.simple.clinic.util.toOptional
+import java.util.UUID
+
+class SettingsEffectHandlerTest {
+
+  private val userSession = mock<UserSession>()
+  private val effectsSubject = PublishSubject.create<SettingsEffect>()
+
+  private val effectHandlerTest: TestObserver<SettingsEvent> = effectsSubject
+      .compose(SettingsEffectHandler.create(
+          userSession = userSession,
+          schedulersProvider = TrampolineSchedulersProvider()
+      ))
+      .test()
+
+  @After
+  fun tearDown() {
+    effectHandlerTest.dispose()
+  }
+
+  @Test
+  fun `user details must be fetched when the load user details event is received`() {
+    // given
+    val savedUser = PatientMocker.loggedInUser(
+        uuid = UUID.fromString("eadfd8ef-6c88-4bc9-ba96-e05bec2d6d8b"),
+        name = "Mahalakshmi Puri",
+        phone = "1234567890"
+    )
+    val savedUserWithUpdatedName = savedUser.copy(fullName = "Mahalakshmi Sharma")
+    val savedUserWithUpdatedPhoneNumber = savedUser.copy(phoneNumber = "1234567891")
+    val savedUserUpdatedNameAndPhoneNumber = savedUser.copy(fullName = "Mahalakshmi T Sharma", phoneNumber = "0987654321")
```

Comment:
We are just illustrating that whenever the underlying stream emits, there will be an update. Why not just do 2 events. More events make the test noisier and developers looking at this example will also pick up the pattern.

Reply:
Makes sense, I'll change this.

Reply:
Done